src/init.cpp

```diff
@@ -235,8 +235,11 @@ void Shutdown(InitInterfaces& interfaces)
     //
     // g_chainstate is referenced here directly (instead of ::ChainstateActive()) because it
     // may not have been initialized yet.
-    if (g_chainstate && g_chainstate->CanFlushToDisk()) {
-        g_chainstate->ForceFlushStateToDisk();
+    {
+        LOCK(cs_main);
```

Comment:
Don't know about this lock, there is no thread which reset `g_chainstate`, and if we shutdown before we init `g_chainstate` we should be still single-threaded. Anyway, taking a lock at shutdown shouldn't hurt

Reply:
Yeah, only added this to satisfy the new lock annotations on `CanFlushToDisk()`.